ci/prow/make_config.go

```diff
@@ -996,9 +1190,308 @@ func (a *stringArrayFlag) Set(value string) error {
+func addTestCoverageJobIfNeeded(jobDetailMap *map[string][]string, repoName string) {
+	if goCoverageMap[repoName] {
+		newJobTypes := append((*jobDetailMap)[repoName], "test-coverage")
+		(*jobDetailMap)[repoName] = newJobTypes
+	}
+}
+
+// generateSection generates the configs for the section with the given generator
+func generateSection(sectionName string, generator testgridEntityGenerator) {
+	outputConfig(sectionName + ":")
+	for _, projName := range projNames {
+		repos := metaData[projName]
+		for _, repoName := range repoNames {
+			if _, exists := repos[repoName]; exists {
+				jobNames := repos[repoName]
+				repoName = buildProjRepoStr(projName, repoName)
+				generator(repoName, jobNames)
+			}
+		}
+	}
+}
+
+// generateTestGroup generates the test group configuration
+func generateTestGroup(repoName string, jobNames []string) {
+	for _, jobName := range jobNames {
+		testGroupName := getTestGroupName(repoName, jobName)
+		gcsLogDir := ""
+		extras := make(map[string]string)
+		switch jobName {
+		case "continuous", "dot-release", "auto-release", "performance", "latency", "api-coverage", "playground":
+			gcsLogDir = fmt.Sprintf("%s/%s/ci-%s-%s", gcsBucket, logsDir, repoName, jobName)
+			if jobName == "playground" {
+				extras["alert_stale_results_hours"] = "168"
+			}
+
+			// TODO: confirm if they are needed or not
+			if jobName == "latency" {
+				extras["short_text_metric"] = "latency"
+			}
+			if jobName == "api-coverage" {
+				extras["short_text_metric"] = "api_coverage"
+			}
+			if jobName == "performance" {
+				extras["short_text_metric"] = "perf_latency"
+			}
+		case "nightly":
+			gcsLogDir = fmt.Sprintf("%s/%s/ci-%s-%s-release", gcsBucket, logsDir, repoName, jobName)
+		case "test-coverage":
+			gcsLogDir = fmt.Sprintf("%s/%s/ci-%s-%s", gcsBucket, logsDir, repoName, "go-coverage")
+			extras["short_text_metric"] = "coverage"
+		default:
+			continue
+		}
+		executeTestGroupTemplate(testGroupName, gcsLogDir, extras)
+	}
+}
+
+// executeTestGroupTemplate outputs the given test group config template with the given data
+func executeTestGroupTemplate(testGroupName string, gcsLogDir string, extras map[string]string) {
+	var data testGroupTemplateData
+	data.Base.TestGroupName = testGroupName
+	data.GcsLogDir = gcsLogDir
+	data.Extras = extras
+	executeTemplate("test group", testGroupTemplate, data)
+}
+
+// generateDashboard generates the dashboard configuration
+func generateDashboard(repoName string, jobNames []string) {
+	outputConfig(fmt.Sprintf("- name: %s\n  dashboard_tab:", repoName))
+	for _, jobName := range jobNames {
+		testGroupName := getTestGroupName(repoName, jobName)
+		baseOptions := "sort-by-name="
+		extras := make(map[string]string)
+		switch jobName {
+		case "continuous":
+			dashboardTabName := jobName
+			executeDashboardTabTemplate(dashboardTabName, testGroupName, baseOptions, extras)
+
+			if repoName == "knative-serving" {
```

Comment:
Add a comment here saying this is a special case for knative/serving, as the conformance-tests tab is just a filtered view of the continuous tab.